feat: add docs for migration to v0.70.1 (#557)
vegaprotocol/documentation: docs/node-operators/migration-guides/upgrade-to-0701.md

cp -r <VEGAVISOR-HOME>/current <BACKUP-FOLDER>/v0.70.1/vegavisor-current

# Check if backup has been successfully done*; check if all files has been copied correctly
tree <BACKUP-FOLDER>

# Backup PostgreSQL if you have been running data node**
pg_dump --host=localhost --port=5432 --username=<VEGA-DB-USER> --password -Fc -f <BACKUP-FOLDER>/v0.70.1/data_node_db.bak.sql <VEGA-DB-NAME>
```

**Notes**: 
* *The `tree` command needs to be installed (e.g. `apt-get install -y tree`) but it is the easiest way to see if backup files match the original files without going too deep into details.
*  **You might see some errors when running `pg_dump`. To learn if they can be safely ignored, see the [troubleshooting section in the official timescaledb docs ↗](https://docs.timescale.com/timescaledb/latest/how-to-guides/backup-and-restore/troubleshooting/).

### 3. Download new binaries
Download new binaries for the upgrade version from the [GitHub releases page ↗](https://github.com/vegaprotocol/vega/releases) and unzip them. Save them in the `<VEGA-BIN>` and the `<VISOR-BIN>` you chose. 

The binaries you need: 
* Vega binary: Also includes Tendermint and data node as binary subcommands

> * Visor binary: Optional for setting up Visor for protocol upgrades (See the docs listed in [step 7](#7-read-the-visor-documentation) for information on Visor.

Author: Ahh yess, fixed, I have referenced to release 0.68.2

# Unzip downloaded archives
unzip vega-linux-amd64.zip
unzip visor-linux-amd64.zip

mv vega <VEGA-BIN>
mv visor <VISOR-BIN>
```

:::node  Manual update for binaries when you are running visor
Visor cannot automatically restart(or upgrade) binaries because We are restarting the network with the `checkpoint`. Once you stop the Vega network, you must update the old binaries with the downloaded ones. You can find the binary path in the `<VEGAVISOR-HOME>/current/run-config.toml` file. Usually the path is `<VEGAVISOR-HOME>/current/vega`
:::

### 4. Reset and clear all data

:::warning Back up files before progressing
Ensure you have a backup of the network files because the steps below will remove data from your home.

You may also risk losing your wallets, so back them up as well.
:::

1. Call unsafe reset all for Tendermint: `<VEGA-BIN> tendermint unsafe-reset-all --home <TENDERMINT-HOME>`
2. Call unsafe reset all for vega core: `<VEGA-BIN> unsafe_reset_all --home <VEGA-NETWORK-HOME>`
3. Remove the data-node state directory: `rm -f <VEGA-NETWORK-HOME>/state/data-node`
4. Recreate the PostgreSQL database if you have data within: 

>     - a. Call the following command in PostgreSQL terminal: `DROP DATABASE IF EXISTS <VEGA-DB-NAME>`
>     - b. Call the following command in PostgreSQL terminal: `CREATE DATABASE <VEGA-DB-NAME> WITH owner=<VEGA-DB-USER>`. 

Reviewer: ```suggestion
    - `DROP DATABASE IF EXISTS <VEGA-DB-NAME>`
    - `CREATE DATABASE <VEGA-DB-NAME> WITH owner=<VEGA-DB-USER>`

```

:::note
If you have to setup postgresql, follow instructions in the step to [Install/Upgrade PostgreSQL instance](./upgrade-node.md#15-installupgrade-postgresql-for-data-node)
:::

> 

Reviewer: Should this be removed? There is no step 15/info on how to do this

Author: The step 15 is in previous migration guide

### 5. Prepare genesis file
We recommend doing this at the beginning of the upgrade procedure, but this can happen at any point before validators start the network. After the genesis is prepared, all the validators must use the same `genesis.json` file.

To load the checkpoint, find more information in the [restart network guide](../how-to/restart-network.md#load-checkpoint) 

1. One of the validators will adjust [the genesis file ↗](https://github.com/vegaprotocol/networks/blob/master/mainnet1/genesis.json) in the [Vega Protocol networks repository ↗](https://github.com/vegaprotocol/networks). You also may ask the vega team member to review it.
2. The person responsible for updating genesis needs to create a PR with changes.
3. All of the validators need to accept changes and approve the PR.
4. Approved PR must be merged by one of the validators.

:::note
Make sure, you have correct values for the following parameters in your genesis: `genesis_time`, `time_iota_ms`, `chain_id`, `network_parameters_checkpoint_overwrite`.
:::

> :::note The checkpoint parameters overwrite

Reviewer: ```suggestion
:::note Checkpoint parameters overwrite

```

> All of the network parameters will be taken from checkpoint except:

Reviewer: ```suggestion
All of the network parameters will be taken from checkpoint except those:

```

1. Download genesis
2. Remove old genesis at `<TENDERMINT-HOME>/config/genesis.json`
3. Save new, downloaded genesis to `<TENDERMINT-HOME>/config/genesis.json`
4. Verify genesis - see example below

#### Example workflow
An example workflow for reviewing the genesis file may look like following:

```bash 
# Download genesis
wget https://raw.githubusercontent.com/vegaprotocol/networks/master/mainnet1/genesis.json

# Move old genesis to a different location
cp <TENDERMINT-HOME>/config/genesis.json <TENDERMINT-HOME>/config/genesis.json.bk

# Copy genesis to its final location
cp ./genesis.json <TENDERMINT-HOME>/config/genesis.json

# Verify genesis
<VEGA-BIN> verify genesis <TENDERMINT-HOME>/config/genesis.json
```

### 7. Update Vega core config

There is no change required in the `vega core` config. If you have to prepare the `vega` config from scratch, use the instruction for (v0.68.2)[(./upgrade-to-0682.md#7-update-vega-core-config)]

### 8. Update Tendermint config

There is no change required in the `tendermint` config. If you have to prepare the `tendermint` config from scratch, use the instruction for (v0.68.2)[(./upgrade-to-0682.md#8-update-tendermint-config)]

> ### 9. Update data-node config
> 
> If you were running node in version `v0.68.0+`, the only think you have to update is the `chain_id` in the `data-node` config. If you need to setup data node from scratch, please follow documentation for [v0.68.0](./upgrade-to-0682.md#9-update-data-node-config)
> 
> :::warning Update `chain_id`
> It is important to update the chain_id for your data node config, otherwise your data node will fail.
> :::

Reviewer: ```suggestion
### 9. Update data node config

If you were running a data node in version `v0.68.0+`, the only thing you have to update is the `chain_id` in the `data-node` config. If you need to set up data node from scratch, please follow documentation for [v0.68.0](./upgrade-to-0682.md#9-update-data-node-config)

:::caution Update `chain_id`
It is important to update the chain ID for your data node config, otherwise your data node will fail.
:::

```
